### PR TITLE
freetype2 & harfbuzz & glib: switch to meson + replace gettext with proxy-libintl

### PR DESCRIPTION
**Pull request by benoit-pierre**
Next step: glib update.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1872)

### CODE REVIEW

**Comment by poire-z** [comment]
So, that's the first PR making meson a hard requirement ? Are we all ok with that?
(I still have it not installed, not minding, just asking.)

File: thirdparty/glib/CMakeLists.txt [hunk 1]
```diff
@@ -3,65 +3,38 @@ list(APPEND PATCH_FILES
...
-        list(APPEND gen_cache_config ac_cv_func_mkostemp=no)
-    endif()
-    if(ANDROID)
-        # disable getpwuid, getgrgid and timegm for > android-12
-        list(APPEND gen_cache_config ac_cv_func_posix_getpwuid_r=no)
-        list(APPEND gen_cache_config ac_cv_func_posix_getgrgid_r=no)
-        list(APPEND gen_cache_config ac_cv_func_getpwuid_r=no)
-        list(APPEND gen_cache_config ac_cv_func_getgrgid_r=no)
-        list(APPEND gen_cache_config ac_cv_func_timegm=no)
-    endif()
```
Comment (benoit-pierre) [inline thirdparty/glib/CMakeLists.txt:28]: That part confused me: the comment about android-12 is from 23985d3, in 2016, so I assume 12 is for the API level? Android 3.1  (HONEYCOMB_MR1, 2011), and not Android 12 (S[now Cone], level 31, 2021). The flags themselves come from 23985d3 and 773db3f66 (via db68822f9 & e579d0eba).

**Comment by benoit-pierre** [comment]
> So, that's the first PR making meson a hard requirement ? Are we all ok with that? (I still have it not installed, not minding, just asking.)

Note:
- recent glib versions only support meson
- harfuzz 9.0 dropped autotools support, and the CMake support is community maintained
- fribidi: still support autotools
- freetype2: support all 3

File: thirdparty/glib/CMakeLists.txt (hunk 1 above)
Comment (Frenzie) [reply thirdparty/glib/CMakeLists.txt:28]: Yup, that's API level 12.

https://developer.android.com/guide/topics/manifest/uses-sdk-element.html#ApiLevels

**Comment by Frenzie** [comment]
> So, that's the first PR making meson a hard requirement ? Are we all ok with that?

I am, yes.

File: thirdparty/glib/CMakeLists.txt (hunk 1 above)
Comment (benoit-pierre) [reply thirdparty/glib/CMakeLists.txt:28]: On that subject, this page has IMHO a better overview: https://apilevels.com/.

File: cmake/CMakeLists.txt [hunk 2]
```diff
@@ -84,8 +84,44 @@ function(setup_project NAME)
...
+    execute_process(COMMAND ${${VAR}} ${VERSION_ARG} OUTPUT_VARIABLE VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+    string(REGEX MATCH "(^|[ \n])([0-9][.0-9]*)([\n ]|$)" VERSION "${VERSION}")
+    message(STATUS "Found ${NAME}: ${${VAR}} (found version \"${CMAKE_MATCH_2}\")")
+endfunction()
+
 # }}}.
 
+# Git. {{{
+
+find_package(Git REQUIRED)
```
Comment (Frenzie) [inline cmake/CMakeLists.txt:104]: This refers to the plain Git executable, not Git development files or libraries?

Comment (benoit-pierre) [reply cmake/CMakeLists.txt:104]: Yes, this is just to get a trace with the git version on setup, and an early error if git is missing in some corner cases (building in a docker mounted volume).

**Comment by Frenzie** [comment]
I suspect `harfbuzz: switch to meson for building` should be amended with a `git rm thirdparty/harfbuzz/soversion.patch`.

**Review by Frenzie — APPROVED** [review]

**Pull request merged**